### ISSUE TITLE
Error after playing music several times with AudioCache

### ISSUE THREAD

**Issue by wenxiangjiang** [bluefireteam/audioplayers#843]
Error after playing music several times with AudioCache.

**Full Description**

By executing the following code, an error occurs when press the button several times.

I found that there are already related issues:

#773 #682 #766 

I think the reason of the issue is at `play` method of `AudioCache`.

```
Future<AudioPlayer> play(
    String fileName, {
    double volume = 1.0,
    bool? isNotification,
    PlayerMode mode = PlayerMode.MEDIA_PLAYER,
    bool stayAwake = false,
    bool recordingActive = false,
    bool? duckAudio,
  }) async {
    String url = await getAbsoluteUrl(fileName);
    AudioPlayer player = _player(mode);
    player.setReleaseMode(ReleaseMode.STOP); // this line
    await player.play(
      url,
      volume: volume,
      respectSilence: isNotification ?? respectSilence,
      stayAwake: stayAwake,
      recordingActive: recordingActive,
      duckAudio: duckAudio ?? this.duckAudio,
    );
    return player;
  }
```

This line was added because #618 . That is a good PR, but it will cause errors when not using fixedPlayer. Maybe due to large amount of `AudioPlayer` will not be released.

I think we should add a judgment like this:

```
Future<AudioPlayer> play(
    String fileName, {
    double volume = 1.0,
    bool? isNotification,
    PlayerMode mode = PlayerMode.MEDIA_PLAYER,
    bool stayAwake = false,
    bool recordingActive = false,
    bool? duckAudio,
  }) async {
    String url = await getAbsoluteUrl(fileName);
    AudioPlayer player = _player(mode);
    
    // this three lines
    if (fixedPlayer != null) {
      player.setReleaseMode(ReleaseMode.STOP);
    }
    
    await player.play(
      url,
      volume: volume,
      respectSilence: isNotification ?? respectSilence,
      stayAwake: stayAwake,
      recordingActive: recordingActive,
      duckAudio: duckAudio ?? this.duckAudio,
    );
    return player;
  }
```

If my idea meets the needs, I'm interested in working on a PR for this.

**Code to Reproduce**

```
// Create a reference
final AudioCache _audioCache = AudioCache(prefix: 'assets/audio/');

// Press several times, an error occurs
TextButton(
  onPressed: () {
    _audioCache.play('demo.mp3');
  },
  child: Text('Play'),
),
```

**Log Errors**

```
D/MediaPlayer( 5024): setSubtitleAnchor in MediaPlayer
E/MediaPlayer( 5024): error (1, -19)
E/MediaPlayer( 5024): Error (1,-19)
E/MediaPlayer( 5024): pause called in state 0, mPlayer(0x83798720)
E/MediaPlayer( 5024): error (-38, 0)
E/MediaPlayer( 5024): Attempt to perform seekTo in wrong state: mPlayer=0x83798720, mCurrentState=0
E/MediaPlayer( 5024): error (-38, 0)
E/MediaPlayer( 5024): Error (-38,0)
E/MediaPlayer( 5024): Error (-38,0)
```

**Files/URLs/Sources**

None.

**Screenshots**

None.

**Platforms**

* OS: Android
* OS version: 7.1.2
* Device: physical device. Internal model.
* flutter version: 2.0.3
* audioplayers version: 0.18.3
* release or not release: release and debug modes.
* does the error occur and does it have any peculiarities: No.

**Comment by erickzanardo** [comment]
This changes seems to make sense for me @wenxiangjiang, would you be willing to put a PR for that?

**Comment by wenxiangjiang** [comment]
@erickzanardo sure. I will do it soon.

**Comment by AhmedNourJamalElDin** [comment]
This still happens to me using the repo as dart dependency i.e. not the published version of the package.

```
audioplayers:
      git:
          url: git://github.com/luanpotter/audioplayers.git
          ref: master
          path: packages/audioplayers
 ```

**Comment by wenxiangjiang** [comment]
@AhmedNourJamalElDin It works well for me. Can you confirm whether the code has been updated to the latest?

**Comment by AhmedNourJamalElDin** [comment]
Yup

Error (1, -19)

**Comment by luanpotter** [comment]
Reopening since it appears there is still some issues (might be fixed by @fdietze 's PR)

**Comment by Marvin10193** [comment]
Any update regarding this issue?

**Comment by tamoyal** [comment]
I'm still getting this issue as well:
```
E/MediaPlayerNative( 9491): error (1, -19)
E/MediaPlayer( 9491): Error (1,-19)
E/MediaPlayerNative( 9491): pause called in state 0, mPlayer(0x7f05f6e0)
E/MediaPlayerNative( 9491): error (-38, 0)
E/MediaPlayerNative( 9491): Attempt to perform seekTo in wrong state: mPlayer=0x7f05f6e0, mCurrentState=0
E/MediaPlayerNative( 9491): error (-38, 0)
2
E/MediaPlayer( 9491): Error (-38,0)
```

I also have not been able to figure out a workaround yet.

**Comment by lyseiha** [comment]
any update yet bro?

**Comment by fdietze** [comment]
Have you tried using this PR? https://github.com/luanpotter/audioplayers/pull/872

**Comment by tamoyal** [comment]
@fdietze Well the maintainer seems to think that's not the issue but I'm certainly going to try it. I noticed the latest version had a potential fix for audio failing after repeat usage but I'm still seeing the errors come in after updating to the latest. I also wonder if just waiting 500ms after setting the url would work. That's ok for my use case, maybe not other people's

**Comment by Marvin10193** [comment]
I havent seen the problem again with the latest update, but no excessive testing has been done yet on my end

**Comment by tamoyal** [comment]
@Marvin10193 I _have_ done extensive testing and still get the issue

**Comment by Marvin10193** [comment]
No, I did not include that PR.

**Comment by Gustl22** [comment]
Can you reconfirm with `audioplayers: ^1.1.0`. This release also includes waiting for the `prepare` statement.

**Comment by Marvin10193** [comment]
Hey,
sorry I am no longer working on the App as it only was semester task and currently I have no time to reconfirm. GL for the future though :)

**Comment by Gustl22** [comment]
Likely fixed with #1191
